datalake/src/main/java/com/sequenceiq/datalake/controller/sdx/SdxBackupController.java

```diff
@@ -38,7 +38,22 @@ public SdxDatabaseBackupResponse backupDatabaseByName(@ResourceName String name,
             sdxDatabaseBackupResponse.setOperationId(backupId);
             return sdxDatabaseBackupResponse;
         } catch (NotFoundException notFoundException) {
-            return sdxBackupRestoreService.triggerDatabaseBackup(sdxCluster, backupId, backupLocation);
+            return sdxBackupRestoreService.triggerDatabaseBackup(sdxCluster, backupId, backupLocation, true);
+        }
+    }
+
+    @Override
+    @CheckPermissionByResourceName(action  = AuthorizationResourceAction.BACKUP_DATALAKE)
+    public SdxDatabaseBackupResponse backupDatabaseByNameConditionallyCloseConnections(@ResourceName String name, String backupId,
+            String backupLocation, boolean closeConnections) {
```

Comment:
Please change the API to accept the Request object with all the fields listed as arguments.
That way we don't have to change the API when we change the arguments to the API.